Block examples documentation

erral opened the pull request:
- [x] I signed and returned the [Plone Contributor Agreement](https://plone.org/foundation/contributors-agreement), and received and accepted an invitation to join a team in the Plone GitHub organization.
- [x] I verified there aren't other open [pull requests](https://github.com/plone/volto/pulls) for the same change.
- [x] I followed the guidelines in [Contributing to Volto](https://6.docs.plone.org/volto/contributing/index.html).
- [x] I succesfully ran [code linting checks](https://6.docs.plone.org/volto/contributing/linting.html) on my changes locally.
- [x] I succesfully ran [unit tests](https://6.docs.plone.org/volto/contributing/testing.html) on my changes locally.
- [x] I succesfully ran [acceptance tests](https://6.docs.plone.org/volto/contributing/acceptance-tests.html) on my changes locally.
- [x] If needed, I added new tests for my changes.
- [x] If needed, I added [documentation](https://6.docs.plone.org/volto/contributing/documentation.html#narrative-documentation) for my changes, either in the Storybook or narrative documentation.
- [x] I included a [change log entry](https://6.docs.plone.org/contributing/index.html#contributing-change-log-label) in my commits.

-----

I have reviewed the block documentation, converted some code blocks from class components to functional components, and provide an example of a simple block.


<!-- readthedocs-preview volto start -->
----
📚 Documentation preview 📚: https://volto--6560.org.readthedocs.build/

<!-- readthedocs-preview volto end -->

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+We can create a block that uses `variations`. A {term}`variation` is an alternative view of a block. This variation is shown as an additional option in the schema editor and lets the webmaster to change how this block is viewed. Think of it as a different view of the same block.
+
+A {term}`schema enhancer` is an option of a variation. Using this schema enhancer, the block schema can be extended to have additional fields.
```
We need to add this term to the `plone/documentation` `glossary.md`.

```markdown
schema enhancer
    A schema enhancer uses the `schemaEnhancer` function, which can modify the schema used by the `InlineForm` component.
    Any registered extension plugin can provide a `schemaEnhancer` function.
    This function receives an object with `formData`, which is the block data; `schema`, which is the original schema to modify; and the injected `intl`, which aids with internationalization.
```

```suggestion
A {term}`schema enhancer` is an option of a variation.
Using this schema enhancer, you can extend the block schema to have additional fields.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+A {term}`schema enhancer` is an option of a variation. Using this schema enhancer, the block schema can be extended to have additional fields.
+
+What we need to do is to define the schema, the view component, the variations, the schema enhancer and configure the block settings.
```
```suggestion
To do so, define the schema, view component, variations, and schema enhancer, then configure the block settings.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+## Preparations
+
+In your volto addon, create a folder inside the {file}`components` folder to save all the files required to create a block.
```
```suggestion
In your Volto add-on, create a subfolder {file}`ExampleBlock06` inside the {file}`components` folder to save all the files required to create a block.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+In your volto addon, create a folder inside the {file}`components` folder to save all the files required to create a block.
+
+Name this folder as {file}`ExampleBlock06`.
```
```suggestion
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+## Schema
+
+Create a {file}`Schema.js` file inside the {file}`ExampleBlock06` folder, with the following contents:
```
```suggestion
Create a {file}`Schema.js` file inside the {file}`ExampleBlock06` folder, with the following contents.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+## Messages
+
+As you have noted, we have prepared the block to be internationalized, {term}`internanationalization` (i18n), is the process of creating user interfaces which are suitable for different languages and cultural contexts.
```
```suggestion
As you may have noted, you prepared the block for internationalization.
{term}`Internationalization` (i18n) is the process of creating user interfaces which are suitable for different languages and cultural contexts.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+As you have noted, we have prepared the block to be internationalized, {term}`internanationalization` (i18n), is the process of creating user interfaces which are suitable for different languages and cultural contexts.
+
+So we need a file {file}`messages.js` in the same {file}`ExampleBlock06` folder with the following contents:
```
```suggestion
Create a file {file}`messages.js` in the same {file}`ExampleBlock06` folder with the following contents.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+## View component
+
+In this case, as we are using variations, the view component needs to use the variation template to render the contents of the block.
```
```suggestion
For variations, the view component needs to use the variation template to render the contents of the block.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+In this case, as we are using variations, the view component needs to use the variation template to render the contents of the block.
+
+This is easily achieved using the `variation` coming on the `props` of the block.
```
```suggestion
You can do so using the `variation` from the `props` of the block.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+This is easily achieved using the `variation` coming on the `props` of the block.
+
+Create a file {file}`View.jsx` in the {file}`ExampleBlock06` folder with the following contents:
```
```suggestion
Create a file {file}`View.jsx` in the {file}`ExampleBlock06` folder with the following contents.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+import React from 'react';
+
+const View = (props) => {
+  // data holds the values entered in the block edit form
```
```suggestion
  // `data` holds the values entered in the block edit form
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+const View = (props) => {
+  // data holds the values entered in the block edit form
+  // className holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`
```
```suggestion
  // `className` holds the CSS class names injected into this block by Volto's `styleClassNameExtenders`
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+const View = (props) => {
+  // data holds the values entered in the block edit form
+  // className holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`
+  // style holds the CSS properties injected to this block by Volto's `Block Sytle Wrapper`
```
```suggestion
  // `style` holds the CSS properties injected into this block by Volto's `Block Style Wrapper`
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+  // data holds the values entered in the block edit form
+  // className holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`
+  // style holds the CSS properties injected to this block by Volto's `Block Sytle Wrapper`
+  // variation holds the variation selected in the block editor, and it is an object as defined in the block configuration
```
```suggestion
  // `variation` holds the variation selected in the block editor, and it is an object as defined in the block configuration
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
````diff
@@ -0,0 +1,274 @@
+```js
+const schemaEnhancerVariation02 = ({ formData, schema, intl }) => {
+  // schema holds the original schema (see the Schema.js file)
+  // so we need to define the new property under `schema.properties`
````
```suggestion
  // so you need to define the new property under `schema.properties`
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+const schemaEnhancerVariation02 = ({ formData, schema, intl }) => {
+  // schema holds the original schema (see the Schema.js file)
+  // so we need to define the new property under `schema.properties`
+  // and push its name to the relevant fieldset, in our case the first one (note the `fieldsets[0]`)
```
```suggestion
  // and push its name to the relevant fieldset, in this case the first one (note the `fieldsets[0]`)
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
````diff
@@ -0,0 +1,274 @@
+
+export default View;
+```
+Create a file {file}`VariationView02.jsx` in the {file}`ExampleBlock06` folder with the following contents:
````
```suggestion

Create a file {file}`VariationView02.jsx` in the {file}`ExampleBlock06` folder with the following contents.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+We need to configure the schema enhancer function. In this case we will be adding a new field named `color` when using the Variation 2.
+
+Create a file {file}`enhancers.js` in the {file}`BlockSchema` folder with the following content:
```
```suggestion
Create a file {file}`enhancers.js` in the {file}`BlockSchema` folder with the following content.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+## Schema enhancer
+
+We need to configure the schema enhancer function. In this case we will be adding a new field named `color` when using the Variation 2.
```
```suggestion
Next you need to configure the schema enhancer function.
In this example, you will add a new field named `color` when using `schemaEnhancerVariation02`.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+## Variations
+
+Now we need to create one or more variations that will be available for this block.
```
```suggestion
Next create one or more variations that will be available for this block.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+Now we need to create one or more variations that will be available for this block.
+
+Create a file {file}`VariationView01.jsx` in the {file}`ExampleBlock06` folder with the following contents:
```
```suggestion
Create a file {file}`VariationView01.jsx` in the {file}`ExampleBlock06` folder with the following contents.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
````diff
@@ -0,0 +1,274 @@
+export default View;
+```
+
+As you can see, in this case the variations are pretty much the same, the only difference is the text that is rendered in the `<h2>` tag. But it can be anything.
````
```suggestion
As you can see, the variations are pretty much the same.
The only difference is the text that is rendered in the `<h2>` tag.
But it can be anything.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+With all the block components ready, you need to register the block into Volto.
+
+To do so, open your addon's {file}`index.js` file, that will have the following contents:
```
```suggestion
To do so, open your add-on's {file}`index.js` file, and add the following contents.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
````diff
@@ -0,0 +1,274 @@
+export default applyConfig;
+```
+
+And before the last `return config;` statement, write the following configuration:
````
```suggestion
Before the last `return config;` statement, write the following configuration.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+      },
+    ],
+  };
+
```
```suggestion
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
````diff
@@ -0,0 +1,274 @@
+
+```
+
+On the top of the file you will need to import the relevant components, as follows:
````
```suggestion
At the top of the file, import the relevant components as follows.
```

stevepiercy commented on docs/source/blocks/examples/customviewvariationsandschemaenhancer.md:
```diff
@@ -0,0 +1,274 @@
+
+Your block is ready to be used in your site.
+
+Restart your Volto site and you will be able to add it using the block add form.
```
```suggestion
Restart your Volto site, and you can add it using the block add form.
```

stevepiercy reviewed (changes requested):
This is great stuff. Thank you!

I created a related PR to define Glossary terms at https://github.com/plone/documentation/pull/1829, which needs to be merged before this PR to pull them into this PR's docs preview.

ichim-david commented:
@erral once you apply @stevepiercy suggestions and you feel that the work is ready for review you can add me as a reviewer if you would like, I volunteer to read the docs and give any potential suggestions if needed.

erral commented on docs/source/blocks/settings.md:
```diff
@@ -48,6 +48,10 @@ const customBlocks = {
     },
+    // A block can have a function that returns the schema or directly the schema
+    // of the block that will be automatically rendered in the sidebar editor
+    // in case that the `edit` setting of the block is set to `null`
```
The `blockSchema` property can either be a schema by itself (a JavaScript object describing the schema), or a function that returns a schema.

stevepiercy replied:
Perfect.

```suggestion
    // The `blockSchema` property can either be a schema by itself
    // (a JavaScript object describing the schema),
    // or a function that returns a schema.
```

erral commented on docs/source/blocks/examples/customviewandvariations.md:
```diff
@@ -0,0 +1,248 @@
+
+# Block with a custom schema and variations
+
+We can create a block that uses `variations`. A {term}`variation` is an alternative view of a block. This variation is shown as an additional option in the schema editor and lets the webmaster to change how this block is viewed. Think of it as a different view of the same block
```
No, in this case it should be the editor. When registering a `variation` we let the editor select how the block will be shown.

stevepiercy replied:
Updated suggestion.

erral commented on docs/source/blocks/examples/customviewandvariations.md:
```diff
@@ -0,0 +1,248 @@
+  );
+};
+
+// the `withBlockExtensions` HOC, makes the variation selector available in the block edit form
```
HOC: High Order Component. We should add a term indicator here.

stevepiercy replied:
Done in 836d0944772159859c9e26ada9af9b8b9e99d411. However, we can't use terms in code blocks, so we need it somewhere in the narrative text.

stevepiercy replied:
Done in https://github.com/plone/volto/pull/6560/commits/3d83b5497c3ee0dd8f2f0709a7c6cce834ad3f3b

erral commented:
Thanks @stevepiercy for your excellent review!

stevepiercy commented on docs/source/blocks/settings.md:
```diff
@@ -48,6 +48,10 @@ const customBlocks = {
       // Required for alternate default block types implementations.
       // See also [Settings reference](/configuration/settings-reference)
     },
+    // A block can have a function that returns the schema or directly the schema
```
```suggestion
```

stevepiercy commented on docs/source/blocks/settings.md:
```diff
@@ -48,6 +48,10 @@ const customBlocks = {
       // See also [Settings reference](/configuration/settings-reference)
     },
+    // A block can have a function that returns the schema or directly the schema
+    // of the block that will be automatically rendered in the sidebar editor
```
```suggestion
```

stevepiercy reviewed (changes requested):
Update comment suggestion

stevepiercy commented:
Direct link to preview.

https://volto--6560.org.readthedocs.build/blocks/examples/custom-schema-and-view.html#see-it-in-action

We need to fix some syntax here.

stevepiercy commented on docs/source/blocks/examples/custom-view-and-variations.md:
```diff
@@ -0,0 +1,251 @@
+      },
+    ],
+  };
+
```
```suggestion
```

stevepiercy commented on docs/source/blocks/examples/custom-view-and-variations.md:
```diff
@@ -0,0 +1,251 @@
+
+## See it in action
+
+Your block is ready to be used in your site.
```
```suggestion
Your block is ready to use in your site.
```

stevepiercy commented on docs/source/blocks/examples/custom-schema-and-view.md:
```diff
@@ -0,0 +1,165 @@
+
+## See it in action
+
+Your block is ready to be used in your site.
```
```suggestion
Your block is ready to use in your site.
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+
+(custom-schema-view-variations-schema-enhancer)=
+
+# Block with a custom schema, variations, and a schema enhancer in a variation.
```
```suggestion
# Block with a custom schema, variations, and a schema enhancer in a variation
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+
+This example builds upon the previous example, {doc}`custom-view-and-variations`, where you can create a block with a custom schema, variations, and a schema enhancer in one of the variations.
+A {term}`variation` is an alternative view of a block.
+The variation is shown as an additional option in the schema editor, and lets the developer change how this block is viewed.
```
```suggestion
The variation is shown as an additional option in the schema editor, and lets the editor change how this block is viewed.
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+import React from 'react';
+
+const View = (props) => {
+  // `data` holds the values entered in the block edit form
```
```suggestion
  // `data` holds the values entered in the block edit form.
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+
+const View = (props) => {
+  // `data` holds the values entered in the block edit form
+  // className holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`
```
```suggestion
  // `className` holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`.
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+const View = (props) => {
+  // `data` holds the values entered in the block edit form
+  // className holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`
+  // `style` holds the CSS properties injected into this block by Volto's `Block Style Wrapper`
```
```suggestion
  // `style` holds the CSS properties injected into this block by Volto's `Block Style Wrapper`.
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+  // `data` holds the values entered in the block edit form
+  // className holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`
+  // `style` holds the CSS properties injected into this block by Volto's `Block Style Wrapper`
+  // `variation` holds the variation selected in the block editor, and it is an object as defined in the block configuration
```
```suggestion
  // `variation` holds the variation selected in the block editor, and it is an object as defined in the block configuration.
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+  );
+};
+
+// the `withBlockExtensions` HOC, makes the variation selector available in the block edit form
```
```suggestion
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+};
+
+// the `withBlockExtensions` HOC, makes the variation selector available in the block edit form
+// and provides the `variation` property in the props.
```
```suggestion
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
````diff
@@ -0,0 +1,280 @@
+// and provides the `variation` property in the props.
+export default withBlockExtensions(View);
+```
+
````
```suggestion

The `withBlockExtensions` {term}`HOC` makes the variation selector available in the block edit form and provides the `variation` property in the props.

```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+
+## See it in action
+
+Your block is ready to be used in your site.
```
```suggestion
Your block is ready to use in your site.
```

stevepiercy reviewed (changes requested):
Just a bunch of nitpicky stuff, except for changing webmaster or developer to editor. I'll merge these. Over to @ichim-david!

stevepiercy commented:
Glossary terms link now.

https://volto--6560.org.readthedocs.build/blocks/examples/custom-view-and-variations.html#view-component

According to the React docs, HOC is no longer used in modern React, so should we use them anymore?

ichim-david commented on docs/source/blocks/examples/custom-view-and-variations.md:
```diff
@@ -0,0 +1,251 @@
+
+const View = (props) => {
+  // `data` holds the values entered in the block edit form.
+  // `className` holds the CSS class names injected into this block by Volto's `styleClassNameExtenders`.
```
@erral it would be great to split the comments in such a way that you can view it without having to use any scrollbars.
See this screen capture in order to understand what i mean i would like not to have to horizontally scroll in order to read what was the suggestion or documentation for the options used in the block config

![comments-visible](https://github.com/user-attachments/assets/1769a935-f9f8-496a-922d-8d2bf4c0cc04)

ichim-david commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,280 @@
+  // so you need to define the new property under `schema.properties`
+  // and push its name to the relevant fieldset, in this case the first one (note the `fieldsets[0]`)
+  schema.properties.color = {
+    title: 'Color',
```
@erral Since you are teaching the users to use the Intl message format, I think it would be worth it to use it also in the schema enhancer example. Therefore, we always teach them to create add-ons the proper way, thinking of internationalization for other users as well.

ichim-david commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
````diff
@@ -0,0 +1,280 @@
+export default View;
+```
+
+As you can see, the variations are pretty much the same.
````
This is wrong since you have created a schema enhancer and you have defined the color property we should have a rendering of this color property so they can clearly see the difference when rendering the first variation and the second variation

ichim-david reviewed (changes requested):
@erral thank you again for taking the time to write these examples.

I think we are missing an example where you also define a custom edit view since we have created the view component, the schema and the schema enhancer.

What we haven't shown yet to the user is how they can also create a custom edit component.
It can also come in another pull request or added by Steve or anyone else I'm not saying it's mandatory for that example to also be created in this pull request but I think it would have been useful to have that example as well.

Again thank you for your work.

erral commented on docs/source/blocks/examples/custom-schema-and-view.md:
```diff
@@ -0,0 +1,165 @@
+
+## Preparations
+
+In your Volto add-on, create a subfolder {file}`ExampleBlock02` inside the {file}`components` folder to save all the files required to create a block.
```
wrll, this comes from examples I have written for my team. There are 6 examples, but I have brought here only 3, the most relevant ones. I have copy-pasted the code, so that's why we have 2, 5 and 6.

I haven't copied the following ones:

- Block just with schema (automatic view and edit)
- Custom edit component (I will add this following your suggestion)
- Block with 'manual' variations: adding the variation field manually in the schema

erral commented on docs/source/blocks/examples/custom-schema-and-view.md:
````diff
@@ -0,0 +1,165 @@
+To do so, open your add-on's {file}`index.js` file, and insert the following contents.
+
+```js
+const applyConfig = (config) => {
````
I will rewrite this, saying to add the block configuration form before the `return config`, that will be easier to understand.

erral commented:
Thanks @ichim-david and @stevepiercy for your comments.

I have made some modifications in the docs:

- I have included the example with a custom edit component
- I have renamed the examples 1,2,3,4 instead of the previous 2, 5 and 6 :sweat_smile: 
- I have changed the block configuration settings part. I have reworded where the configuration should be included to be clearer for the reader.

stevepiercy commented on docs/source/blocks/examples/custom-schema-view-and-edit.md:
```diff
@@ -0,0 +1,210 @@
+
+Your block is ready to be used in your site.
+
+Restart your Volto site, and now you can add the new block from the block chooser found in the `edit` or `add` new content views.
```
I'm gonna make this silver fancy dancing pants. 🪩 🕺 

```suggestion
Restart your Volto site, and now you can add the new block from the block chooser found in the {guilabel}`Edit` <img alt="Edit icon" src="../../_static/pen.svg" class="inline"> or {guilabel}`Add` <img alt="Add icon" src="../../_static/add-document.svg" class="inline"> content views.
```

stevepiercy reviewed (changes requested):
More nitpicky stuff. I'll merge these.

stevepiercy commented on docs/source/blocks/examples/custom-schema-and-view.md:
```diff
@@ -0,0 +1,153 @@
+
+Your block is ready to be used in your site.
+
+Restart your Volto site, and now you can add the new block from the block chooser found in the `edit` or `add` new content views.
```
```suggestion
Restart your Volto site, and now you can add the new block from the block chooser found in the {guilabel}`Edit` <img alt="Edit icon" src="../../_static/pen.svg" class="inline"> or {guilabel}`Add` <img alt="Add icon" src="../../_static/add-document.svg" class="inline"> content views.
```

stevepiercy commented on docs/source/blocks/examples/custom-view-and-variations.md:
```diff
@@ -0,0 +1,240 @@
+
+Your block is ready to be used in your site.
+
+Restart your Volto site, and you can add it using the block add form.
```
```suggestion
Restart your Volto site, and now you can add the new block from the block chooser found in the {guilabel}`Edit` <img alt="Edit icon" src="../../_static/pen.svg" class="inline"> or {guilabel}`Add` <img alt="Add icon" src="../../_static/add-document.svg" class="inline"> content views.
```

stevepiercy commented on docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md:
```diff
@@ -0,0 +1,277 @@
+
+Your block is ready to use in your site.
+
+Restart your Volto site, and you can add it using the block add form.
```
```suggestion
Restart your Volto site, and now you can add the new block from the block chooser found in the {guilabel}`Edit` <img alt="Edit icon" src="../../_static/pen.svg" class="inline"> or {guilabel}`Add` <img alt="Add icon" src="../../_static/add-document.svg" class="inline"> content views.
```

stevepiercy reviewed (changes requested):
I'm waiting for the RTD PR preview to complete building before committing these suggestions.

stevepiercy reviewed (approved):
This looks great. Thank you @erral for sharing your work, and @ichim-david for the suggesstions for improvement. ✋ 

@ichim-david one more review from you, and if you approve, please go ahead and merge.

ichim-david reviewed (approved):
@erral and @stevepiercy. I built locally and the code examples look good now without any extra
overflow scrollbars and the build is passing.
Again thank you erral for this work.

davisagli commented:
@erral @ichim-david @stevepiercy Nice work!
